### PR TITLE
Port Interconnection View and the 3 Behavior views to diagram_draw (spec42 #176 phase 2)

### PR DESCRIPTION
**Pull request by jevontech**
## Summary

Phase 2 of #176: extends the `diagram_draw` crate (started in phase 1 / #177, General View only)
to draw the four remaining view types natively in Rust, byte-for-byte identical to the existing
TS/D3 pipeline. Per prior discussion, this lands as a single PR covering all four views rather
than one PR per view.

- **Sequence View** (`sequence.rs`): lifelines, messages (normal/self/reply), activations, fragments.
- **State-Transition View** (`state_transition.rs`): states, composite states with nested regions,
  self-loop transitions, entry/do/exit/guard/effect/accept/send label composition.
- **Action-Flow View** (`action_flow.rs`): actions with input/output parameters, decision/fork/join/
  initial/final nodes, swimlanes, succession/conditional/streaming flows.
- **Interconnection View** (`ibd_*.rs`): parts/containers, ports + the port-overlay relocation pass,
  ELK-route-point resolution with offset scoring, view frame, dual attr/style marker application.

Each view is verified by a new full-tree XML-comparison test (`*_parity.rs`) against a real
TS/D3-rendered SVG fixture dumped from `vscode/diagram-renderer` — the same methodology phase 1
used, chosen deliberately after phase 1's class-count-only tests missed real fidelity bugs that
review later caught.

As in phase 1, this is purely additive: no changes to `headless_renderer.rs`, the LSP, or the
webview. TS/D3 remains the live rendering path; retiring it is a later phase of #176.

## Notable bugs found and fixed while porting

- `contentBounds` (render/export.ts) always uses General View's 200/70 node-size fallback
  defaults for the final SVG viewBox and the IBD view-frame bounds, regardless of view type — IBD's
  280/140 defaults are only used inside `drawInterconnectionContainers`'s own fallback branch.
- `attachBehaviorNodeClick` always sets `style("cursor", "")` in headless export (no click handler
  wired), and `VirtualStyle.setProperty` only omits a declaration on `null`/`undefined`, not on an
  empty string — so `style="cursor: ;"` is real output that must be reproduced, not dropped.

## Test plan

- [x] `cargo test -p diagram_draw` (8/8 passing, including 4 new `*_parity` byte-comparison tests)
- [x] `cargo clippy -p diagram_draw --all-targets` clean
- [x] `cargo build --workspace` / `cargo clippy --workspace --all-targets --all-features -- -D warnings` clean
- [x] `cargo test -p sysml_query --test syntax_authority` clean (7 new EXEMPTIONS across both commits)
- [x] `cargo fmt --all -- --check` clean
- [x] `npx vitest run` in `vscode/diagram-renderer` (351/351 passing)
- [x] `npm run typecheck` in `vscode/diagram-renderer` clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/diagram_draw/src/sequence.rs [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+                    .text(label),
+            );
+        }
+    }
+    root = root.child(message_layer);
+
+    let position_for = |id: &str| -> Option<&MessagePosition> {
+        message_positions
+            .iter()
+            .find(|(candidate, _)| candidate == id)
```
Comment (jevontech) [inline crates/diagram_draw/src/sequence.rs]: The TypeScript `Map` overwrites an earlier position when `messageRef` repeats, but this `Vec` plus `find()` returns the first occurrence. Repeated message names or labels therefore make activations and fragments reference the wrong row. Use map insertion semantics or search from the end.

File: crates/diagram_draw/src/ibd_edges.rs [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+                end,
+                horizontal: (start.y - end.y).abs() < 1e-6,
+                length: (end.x - start.x).hypot(end.y - start.y),
+            }
+        })
+        .collect();
+    let longest_horizontal = segments
+        .iter()
+        .filter(|s| s.horizontal && s.length >= 24.0)
+        .max_by(|a, b| a.length.total_cmp(&b.length));
```
Comment (jevontech) [inline crates/diagram_draw/src/ibd_edges.rs]: The TypeScript stable descending sort selects the first segment when lengths tie, whereas `Iterator::max_by` selects the last equal maximum. Symmetric routes can therefore move their label to another segment. Include the original index as an earlier-wins tie-breaker.

File: crates/diagram_draw/src/ibd_node.rs [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    /// order `renderIbdNode` appends them). Whether to add the `ibd-container` class to that outer
+    /// group is `nodes.rs`'s own `is_ibd_layout_container(node)` check -- the same condition
+    /// `renderIbdNode` computes independently in the original, so it isn't duplicated here.
+    pub children: Vec<Element>,
+    /// `port-icon`/`port-label` pairs, kept separate for the port-overlay relocation step.
+    pub ports: Vec<Element>,
+}
+
+fn truncate(value: &str, max: usize) -> String {
+    let chars: Vec<char> = value.chars().collect();
```
Comment (jevontech) [inline crates/diagram_draw/src/ibd_node.rs]: JavaScript counts UTF-16 code units, while `chars()` counts Unicode scalar values, so labels containing emoji truncate differently from the source renderer. Reuse the `encode_utf16`-based approach used by `behavior_common::truncate_label`; `ibd_edges.rs` has the same new mismatch.

**Review by jevontech — COMMENTED** [review]
Found three renderer-parity issues that should be addressed.

**Comment by jevontech** [comment]
Fixed all three in c8db822a, each with a regression test since none of the existing golden fixtures happened to exercise these edge cases:

- `sequence.rs`: extracted `last_position_for` and search from the end, matching the TS `Map`'s last-wins overwrite semantics on a repeated `messageRef`.
- `ibd_edges.rs`: `interconnectionEdgeLabelAnchor`'s segment selection now breaks length ties on `Reverse(index)`, matching the stable-sort's first-wins tie-break.
- `ibd_node.rs`/`ibd_edges.rs`: both `truncate` helpers now count UTF-16 code units via `encode_utf16`, matching JS `String.length`/`.slice()`, same approach already used by `behavior_common::truncate_label`.

CI is running on the new commit.
